src/mongocxx/test/options/find_one_and_delete.cpp

```diff
@@ -0,0 +1,23 @@
+#include "catch.hpp"
+#include "helpers.hpp"
+
+#include <bsoncxx/builder/stream/document.hpp>
+#include <mongocxx/options/find_one_and_delete.hpp>
+
+using namespace mongocxx;
+
+TEST_CASE("find_one_and_delete", "[find_one_and_delete][option]") {
+    options::find_one_and_delete opts{};
+
+    std::int64_t ms{400};
+    bsoncxx::builder::stream::document proj{};
```

Comment:
I might inline these with finalize and call .view() in CHECK_OPTIONAL_ARGUMENT.

That or add equality operators to compare view's and values to to document::view/value and array::view/value